src/Orleans/Cancellation/GrainCancellationToken.cs

```diff
@@ -56,16 +61,19 @@ public sealed class GrainCancellationToken : IDisposable
 
         internal Task Cancel()
         {
-            _cancellationTokenSource.Cancel();
+            // propagate the exception from the _cancellationTokenSource.Cancel back to the caller
+            // but also cancel _targetGrainReferences. 
+            Task task = OrleansTaskExtentions.WrapInTask(_cancellationTokenSource.Cancel);
```

Comment:
The point here is to add this promise to WhenAll list. 
How can I do this without using TCS? And if I am using TCS, then how what I wrote any different than using TCS directly?